### PR TITLE
Add PM2_ENABLED flag to Docker

### PR DESCRIPTION
**Pull request by jason-fox**
This PR is a potential fix for https://github.com/telefonicaid/iotagent-node-lib/issues/790 - if accepted I can roll out similar PRs across other IoT Agents

* Update entrypoint to use/not use `pm2`
* Tidy whitespace in `Dockerfile`
* Remove anchors mucking up navigation in RTD docs
* Add `PM2_ENABLED` to Docker README
* Update CNR.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -153,6 +153,19 @@ COPY . /opt/iotajson/
...
+The IoT Agent within the Docker image can be run encapsulated within the [pm2](http://pm2.keymetrics.io/) Process
+Manager by adding the `PM2_ENABLED` environment variable.
+
+```console
+docker run --name iotagent -e PM2_ENABLED=true -d fiware/iotagent-json
+```
+
+Use of pm2 is **disabled** by default. It is unnecessary and counterproductive to add an additional process manager if
+your dockerized environment is already configured to restart Node.js processes whenever they exit (e.g. when using
+Kubenetes)
```
Comment (fgalan) [inline docker/README.md]: Typo: Kubernetes

File: rpm/SPECS/iotajson.spec [hunk 2]
```diff
@@ -172,7 +172,7 @@ fi
 * Tue Aug 13 2019 Fermin Galan <fermin.galanmarquez@telefonica.com> 1.11.0
 - Set Nodejs 8 as minimum version in packages.json (effectively removing Nodev6 from supported versions)
 - Add: Reconnect when MQTT closes connection (including mqtt retries and keepalive conf options)
-- Upgrade iotagent-node-lib dependency from 2.8.1 to 2.9.0
+- Upgrade iotagent-node-lib dependency from 2.9.0 to 2.10.0
```
Comment (fgalan) [inline rpm/SPECS/iotajson.spec]: This seems to be already in master (see current https://github.com/telefonicaid/iotagent-json/blob/master/rpm/SPECS/iotajson.spec). Maybe the branch is out of sync and once gets updated with mater this diff will disappear.

Comment (jason-fox) [reply rpm/SPECS/iotajson.spec]: I've rebased the change to clean up the merge conflict. Fixed 998c6635

File: docker/README.md (hunk 1 above)
Comment (jason-fox) [reply docker/README.md]: Fixed d4ecf092f

**Comment by fgalan** [comment]
Check with @AlvaroVega. Ready to merge.

**Comment by AlvaroVega** [comment]
LGTM

**Pull request merged**